tests/datasource/fluent/data_asset/test_file_path_asset.py

```diff
@@ -87,3 +102,319 @@ def test_get_batch_list_from_batch_request__sort_descending(
     for i, batch in enumerate(batches):
         assert batch.metadata["year"] == str(expected_years[i])
         assert batch.metadata["month"] == str(expected_months[i])
+
+
+@pytest.fixture
+def datasource(mocker):
+    return mocker.Mock(spec=PandasFilesystemDatasource)
+
+
+@pytest.fixture
+def file_path_data_connector(mocker):
+    return mocker.Mock(spec=FilePathDataConnector)
+
+
+@pytest.fixture
+def asset(request, datasource, file_path_data_connector) -> _FilePathDataAsset:
+    # asset = _FilePathDataAsset(name="test_asset", type="_sql_asset")
+    asset = request.param
+    # since we're parametrizing these tests multiple ways, this object
+    # will likely be reused, so we make sure the state we care about is reset:
+    asset.batch_definitions = []
+    asset._datasource = datasource  # same pattern Datasource uses to init Asset
+    asset._data_connector = file_path_data_connector
+    return asset
+
+
+PATH_NAME = "data_2022-01.csv"
+
+
+def _asset_parameters():
+    return [
+        pytest.param(
+            _FilePathDataAsset(name="test_asset", type="_sql_asset"), id="FilePathDataAsset"
+        ),
+    ]
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "path",
+    [
+        pytest.param(PATH_NAME, id="String Path"),
+        pytest.param(pathlib.Path(PATH_NAME), id="Pathlib Path"),
+    ],
+)
+@pytest.mark.parametrize("asset", _asset_parameters(), indirect=["asset"])
+def test_add_batch_definition_fluent_file_path__add_batch_definition_path_success(
+    datasource,
+    asset,
+    path: PathStr,
+    file_path_data_connector,
+):
+    # arrange
+    name = "batch_def_name"
+    expected_regex = re.compile(str(path))
+    expected_batch_definition = BatchDefinition(
+        name=name, partitioner=PartitionerPath(regex=expected_regex)
+    )
+    datasource.add_batch_definition.return_value = expected_batch_definition
+    file_path_data_connector.get_matched_data_references.return_value = [PATH_NAME]
+
+    # act
+    batch_definition = asset._add_batch_definition_path(name=name, path=path)
+
+    # assert
+    assert batch_definition == expected_batch_definition
+    datasource.add_batch_definition.assert_called_once_with(expected_batch_definition)
```

Comment:
We check this because we've mocked out the datasource? It seems like it would be testing behavior more/implementation less if we could retrieve the batch definitions from the asset and see it's what we want. I'm ok with this.

Reply:
yeah, the `add_batch_definition` api is well-tested, so i just wanted to keep the scope of the test lean and just make sure we're calling into it correctly.